### PR TITLE
Set DynamicUser= to resolved and networkd

### PR DESCRIPTION
**Pull request by yuwata**
This sets `DynamicUser=` to systemd-resolved.service and systemd-networkd.service.

### CODE REVIEW

File: tmpfiles.d/systemd.conf.m4 [hunk 1]
```diff
@@ -16,11 +16,6 @@ d /run/systemd/sessions 0755 root root -
 d /run/systemd/users 0755 root root -
 d /run/systemd/machines 0755 root root -
 d /run/systemd/shutdown 0755 root root -
-m4_ifdef(`ENABLE_NETWORKD',
-d /run/systemd/netif 0755 systemd-network systemd-network -
-d /run/systemd/netif/links 0755 systemd-network systemd-network -
-d /run/systemd/netif/leases 0755 systemd-network systemd-network -
-)m4_dnl
```
Comment (poettering) [inline tmpfiles.d/systemd.conf.m4]: hmm, i am not sure about this one... we have this API where people can install inotify watches on /run/systemd/netif/links to get notified about network config changes. Currently that works fine already relatively early at boot, as we create the dirs to watch with tmpfiles early on. But this is not going to work if we don't anymore...

File: src/core/execute.c [hunk 2]
```diff
@@ -2089,9 +2089,7 @@ static int setup_exec_directory(
                         }
                 } else {
                         r = mkdir_label(p, context->directories[type].mode);
-                        if (r == -EEXIST)
-                                continue;
-                        if (r < 0)
+                        if (r < 0 && r != -EEXIST)
```
Comment (poettering) [inline src/core/execute.c:2092]: this undoes a central part of 30c81ce2cef97b05e143c7adf4cd1b1c5fb59932.

@keszybz can you comment?

Comment (yuwata) [reply src/core/execute.c:2092]: Oh, I forgot the commit. Hmm. So, how should we migrate runtime directories? Should we add a flag to `RuntimeDirectory=`?

Comment (yuwata) [reply src/core/execute.c:2092]: What do you think about `chown`ing runtime directories only when `DynamicUser=` is set?

Comment (keszybz) [reply src/core/execute.c:2092]: I think this is OK. In 30c81ce2cef97b05e143c7adf4cd1b1c5fb59932 I didn't think about this case (runtime directory of a dynamic user when this user gets a new number).

**Comment by keszybz** [comment]
Hmm, I don't understand how this is supposed to work: if persistent files are owned by a dynamic user, and the dynamic user may change, how do we know that a different service that uses dynamic user too does not accidentally get access to `/run/systemd/netif/links`?

**Comment by yuwata** [comment]
The situation seems to be similar to #5394 or #7090.

**Pull request merged**